ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java

```diff
@@ -364,6 +364,9 @@ TruncationSegments purge(long index) {
           list.addAll(segments);
           segments.clear();
           sizeInBytes = 0;
+        } else if (segmentIndex == -1) {
```

Comment:
When segments is Empty, it seems code will go to `if (segmentIndex == -segments.size() - 1) {` ?